File: src/main/java/org/hwabeag/divesystem/events/MoveEvent.java
package org.hwabeag.divesystem.events;

import org.bukkit.ChatColor;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;
import org.hwabeag.divesystem.config.ConfigManager;

import java.util.Objects;

public class MoveEvent implements Listener {
    FileConfiguration DiveSystemConfig = ConfigManager.getConfig("dive-system");
    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");
    String Prefix = ChatColor.translateAlternateColorCodes('&', (String) Objects.requireNonNull(DiveSystemConfig.getString("dive-system.prefix")));

    @EventHandler
    public void onPlayerMove(PlayerMoveEvent event) {
        Player player = event.getPlayer();
        String name = player.getName();
        if (!PlayerConfig.getBoolean(name + ".잠수정보")) return;
        PlayerConfig.set(name + ".잠수정보", false);
        ConfigManager.saveConfigs();
        player.sendActionBar(Prefix + " 움직임이 있어 잠수가 취소되었습니다.");
    }
}

File: src/main/kotlin/org/hwabeag/divesystem/events/MoveEvent.kt
package org.hwabeag.divesystem.events

import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerMoveEvent
import org.hwabeag.divesystem.config.ConfigManager
import org.hwabeag.divesystem.util.MessageUtil

class MoveEvent : Listener {
    @EventHandler
    fun onPlayerMove(event: PlayerMoveEvent) {
        val player = event.player
        val playerConfig = ConfigManager.getConfig("player")
        val name = player.name
        if (!playerConfig.getBoolean("$name.잠수정보")) {
            return
        }
        playerConfig.set("$name.잠수정보", false)
        ConfigManager.saveConfigs()
        player.sendActionBar("${MessageUtil.prefix()} 움직임이 있어 잠수가 취소되었습니다.")
    }
}
